### PR TITLE
Skip solver at install time

### LINKED ISSUE #541: RFC: Disabling solves at install time
As of constructor v3, the installers relies on a bundled version of `conda-standalone` to perform the extraction, linking and post-link tasks of the package installation. Using a full conda executable instead of a light reimplementation brings several benefits to the table:

- compatibility with .tar.bz2 and .conda packages
- proper handling of noarch
- topological execution of pre and post link/activation scripts
- battle tested code

However, the [current invocation](https://github.com/conda/constructor/blob/main/constructor/header.sh#L491) has an unintended side-effect: a full solve is performed (again) during install time, even if the input file already contains the "environment solution". 

This behaviour is responsible for at least these two issues: #319 and #456 

I see three solutions here, ordered in growing complexity of implementation:

* Add `--no-deps` to the command. I _think_ this still performs a full solve and might be slower than the options below.
* Use a `@EXPLICIT` input file with relative local paths, instead of "exact specs" (e.g. `pkg=1.2.3=h123abc_0`.
* Rely on a frozen `conda-lock` executable? 😎 Constructor essentially locks an environment and bundles it, putting a OS-specific UI/CLI on top.

I would like to gather your ideas/feedback before we start working on a solution, @conda/constructor 

Thanks!

### PR DESCRIPTION
**Pull request by jaimergp**
### Description

Comes from and closes #541 (read the issue for more details)

I checked using just `--no-deps` as an install flag, and while the docs hint that it skips the dependency checks... what it actually does is to perform a full solve, and then trim the dependencies from the graph.

So instead I am opting for a feature already present in `conda`: `@EXPLICIT` installs using the remote URLs. Since the repodata and channel cache is mocked already to make the previous method work, we get almost everything for free.

Basically, constructor invokes `conda install --offline --file <file>`, with this file:

```
tzdata==2022a=hda174b7_0
ca-certificates==2022.07.19=hca03da5_0
libffi==3.4.2=hc377ac9_4
openssl==1.1.1q=h1a28f6b_0
zlib==1.2.12=h5a0b063_2
setuptools==63.4.1=py310hca03da5_0
certifi==2022.6.15=py310hca03da5_0
sqlite==3.39.2=h1058600_0
pip==22.1.2=py310hca03da5_0
ncurses==6.3=h1a28f6b_3
readline==8.1.2=h1a28f6b_1
python==3.10.4=hbdb9e5c_0
xz==5.2.5=h1a28f6b_1
wheel==0.37.1=pyhd3eb1b0_0
tk==8.6.12=hb8d0fd4_0
libcxx==12.0.0=hf6beb65_1
bzip2==1.0.8=h620ffc9_4
```

What we are doing in this PR is changing the file syntax to:

```
# platform: osx-arm64
@EXPLICIT
https://conda.anaconda.org/pkgs/main/noarch/tzdata-2022a-hda174b7_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/ca-certificates-2022.07.19-hca03da5_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/libffi-3.4.2-hc377ac9_4.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/openssl-1.1.1q-h1a28f6b_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/zlib-1.2.12-h5a0b063_2.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/setuptools-63.4.1-py310hca03da5_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/certifi-2022.6.15-py310hca03da5_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/sqlite-3.39.2-h1058600_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/pip-22.1.2-py310hca03da5_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/ncurses-6.3-h1a28f6b_3.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/readline-8.1.2-h1a28f6b_1.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/python-3.10.4-hbdb9e5c_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/xz-5.2.5-h1a28f6b_1.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/noarch/wheel-0.37.1-pyhd3eb1b0_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/tk-8.6.12-hb8d0fd4_0.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/libcxx-12.0.0-hf6beb65_1.tar.bz2#<md5hash>
https://conda.anaconda.org/pkgs/main/osx-arm64/bzip2-1.0.8-h620ffc9_4.tar.bz2#<md5hash>
```

This still goes through the transaction API, and the records are sorted in topological order, so we should be good.

### Checklist - did you ...

- [X] Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [x] Add / update outdated documentation?

### CODE REVIEW

**Comment by hmaarrfk** [comment]
Does this allow users to update their environments in the future?

**Comment by jaimergp** [comment]
It's like installing a conda-locked environment from scratch.

With the `exclude` option enabled, the environment will be deemed "inconsistent" (same as when you do `conda remove --force <pkg>`) and a further operation on the conda environment might bring it back, though.

**Review by hoechenberger — APPROVED** [review]
LGTM! Very clever approach

**Comment by hmaarrfk** [comment]
Do you think this will solve issues with users having an existing `.condarc` file that specifies different channels? We've been having lots of complaints at Miniforge regarding old installations stopping new installs due to lingering .condarc files

**Comment by jaimergp** [comment]
Yes, it will alleviate solver-derived problems because this solution bypasses it entirely. It might still suffer from other configuration keys, but I can't think of any examples right now.

We do need better condarc isolation for conda-standalone in general, but that's separate issue (https://github.com/conda/constructor/issues/542) we need to solve at the conda level first (my idea is to expose the CONDARC_SEARCH_PATH constant as a context key).

**Comment by jaimergp** [comment]
Any other comments @conda/constructor? I will merge later this week.

File: examples/noconda/construct.yaml [hunk 1]
```diff
@@ -5,6 +5,8 @@ channels:
   - http://repo.anaconda.com/pkgs/main/
 specs:
   - python
+exclude:  # [unix]
+  - tk  # [unix]
```
Comment (hmaarrfk) [inline examples/noconda/construct.yaml:9]: this is interesting as a use case.

**Review by hmaarrfk — APPROVED** [review]

**Review by goanpeca — APPROVED** [review]

**Comment by jaimergp** [comment]
Thanks everybody!

**Comment by EisW** [comment]
Is it possible to ping #656? 
Meanwhile I signed the CLA, but do not know the next step ...

**Pull request merged**